compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -905,6 +983,19 @@ void AMDAIELowerToAIEPass::runOnOperation() {
           applyPatternsAndFoldGreedily(getOperation(), std::move(patterns)))) {
     return signalPassFailure();
   }
+
+  // All Ukernel related function declarations will be within aie.device, so
+  // delete the ones outside from the SymbolTable.
+  SmallVector<func::FuncOp> eraseCandidateFuncOps;
+  getOperation()->walk([&](func::FuncOp funcOp) {
+    if (funcOp.isPrivate() && !funcOp->getParentOfType<AIE::DeviceOp>()) {
+      eraseCandidateFuncOps.push_back(funcOp);
```

Comment:
Isn't it possible to erase the func here?

Reply:
So, this was based on a few similar tryst where I tried deleting/erasing the op being `walk`ed upon and for some reason it crashed.
I've since then tried capturing the candidates and then erasing them afterwards.

I'd indeed prefer what you're suggesting and will give that a try and report here.

Reply:
Done. It worked. :')